File: org.summer.view.widget/src/org/summer/view/widget/TemplateBindingExpression.java
package org.summer.view.widget;

import org.summer.view.widget.data.Expression;
import org.summer.view.widget.TemplateBindingExtension;
/// <summary>
/// A TemplateBinding is used in templates (e.g. ControlTemplate).  When the content of
/// a template is instantiated for inspection, the template bindings are represented with 
///  a TemplateBindingExpression.  (In this case, the expression returns the property's default
/// value.) 
/// </summary> 

//[TypeConverter(typeof(TemplateBindingExpressionConverter))] 
public class TemplateBindingExpression extends Expression
{
    private TemplateBindingExtension _templateBindingExtension;

    /*internal*/ public TemplateBindingExpression( TemplateBindingExtension templateBindingExtension )
    { 
        _templateBindingExtension = templateBindingExtension; 
    }


    /// <summary>
    /// Constructor for TemplateBindingExpression
    /// </summary> 
    public TemplateBindingExtension TemplateBindingExtension
    { 
        get { return _templateBindingExtension; } 
    }


    /// <summary>
    ///     Called to evaluate the Expression value
    /// </summary> 
    /// <param name="d">DependencyObject being queried</param>
    /// <param name="dp">Property being queried</param> 
    /// <returns>Computed value. Default (of the target) if unavailable.</returns> 
    /*internal*/ public /*override*/ Object GetValue(DependencyObject d, DependencyProperty dp)
    { 
        return dp.GetDefaultValue(d.DependencyObjectType);
    }


}

File: summer/WebContent/js/system/windows/TemplateBindingExpression.js
/**
 * TemplateBindingExpression
 */

define(["dojo/_base/declare", "system/Type", "windows/Expression", ], function(declare, Type, Expression){
	var TemplateBindingExpression = declare("TemplateBindingExpression", Expression,{
		constructor:function(/*TemplateBindingExtension*/ templateBindingExtension )
        { 
            this._templateBindingExtension = templateBindingExtension; 
        },
        
        /// <summary>
        ///     Called to evaluate the Expression value
        /// </summary> 
        /// <param name="d">DependencyObject being queried</param>
        /// <param name="dp">Property being queried</param> 
        /// <returns>Computed value. Default (of the target) if unavailable.</returns> 
//        internal override object 
        GetValue:function(/*DependencyObject*/ d, /*DependencyProperty*/ dp)
        { 
            return dp.GetDefaultValue(d.DependencyObjectType);
        }
	});
	
	Object.defineProperties(TemplateBindingExpression.prototype,{
        /// <summary>
        /// Constructor for TemplateBindingExpression
        /// </summary> 
//        public TemplateBindingExtension 
        TemplateBindingExtension:
        { 
            get:function() { return this._templateBindingExtension; } 
        }
	});
	
	TemplateBindingExpression.Type = new Type("TemplateBindingExpression", TemplateBindingExpression, [Expression.Type]);
	return TemplateBindingExpression;
});
